modules/cluster-management/src/main/java/org/apache/ignite/internal/cluster/management/ClusterInitializer.java

```diff
@@ -114,15 +121,17 @@ private CompletableFuture<Void> invokeMessage(Collection<ClusterNode> nodes, Net
                         .invoke(node, message, 10000)
                         .thenAccept(response -> {
                             if (response instanceof InitErrorMessage) {
-                                throw new InitException(String.format(
-                                        "Got error response from node \"%s\": %s", node.name(), ((InitErrorMessage) response).cause()
-                                ));
-                            }
-
-                            if (!(response instanceof InitCompleteMessage)) {
-                                throw new InitException(String.format(
-                                        "Unexpected response from node \"%s\": %s", node.name(), response.getClass()
-                                ));
+                                var errorResponse = (InitErrorMessage) response;
+
+                                throw new InternalInitException(
+                                        String.format("Got error response from node \"%s\": %s", node.name(), errorResponse.cause()),
+                                        errorResponse.shouldCancel()
+                                );
+                            } else if (!(response instanceof InitCompleteMessage)) {
```

Comment:
These options are mutually exclusive, so I don't understand the question. Can you elaborate?

Reply:
If a message is neither an `InitErrorMessage` and also does not satisfy the `!(response instanceof InitCompleteMessage)` condition, then it can only mean one thing - this is an `InitCompleteMessage` in which case the procedure has completed successfully and we don't need to do anything

Reply:
Yep, that definitely has sense.